<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" href="/HiTSeekR/stylesheets/shiori.css">
    <link href='//fonts.googleapis.com/css?family=Montserrat:700,400' rel='stylesheet' type='text/css'>
    <link href='//fonts.googleapis.com/css?family=Merriweather:400,400italic,700,700italic' rel='stylesheet' type='text/css'>
    <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css" rel="stylesheet">
    <link rel="canonical" href="http://nanocan.github.io/HiTSeekR/tutorial/">
    <link rel="shortcut icon" href="/HiTSeekR/favicon.ico">
    <title>Tutorial | HiTSeekR</title>
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
      <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->
    <!-- Additions to <head> tags go here. -->

  </head>
  <body>
  
    <div class="navbar navbar-inverse navbar-static-top">
  
      <div class="container">
        <div class="navbar-header">
          <div class="navbar-toggle-wrapper visible-xs">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".js-navbar-collapse">
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
            </button>
          </div>
          <a href="/HiTSeekR/" class="navbar-brand" style="margin-bottom:100px;"><img src="/HiTSeekR/images/HiTSeekR.png" alt=""></a>
        </div>
        <div class="collapse navbar-collapse js-navbar-collapse">
          <ul class="navbar-nav nav" data-purplecoat="shiori" data-purplecoat-label="nav.html">
            <li><a href="/HiTSeekR/archive">Archive</a></li>
<li><a href="/HiTSeekR/tutorial">Tutorial</a></li>


          </ul>
          <ul class="navbar-nav nav navbar-right" data-purplecoat="shiori" data-purplecoat-label="nav-right.html">
            <li class="hidden-xs"><iframe src="http://ghbtns.com/github-btn.html?user=nanocan&repo=HiTSeekR&type=watch&count=true&size=large"
  allowtransparency="true" frameborder="0" scrolling="0" width="170" height="30" style="margin-top: 14px;"></iframe>
</li>

          </ul>
        </div>
      </div>
    </div>
    <div class="container">
      <div class="row">
        
          <div class="col-sm-8">
            
<h2 id="introduction">Introduction</h2>
<p><a name="Introdcution"></a></p>

<p>High-throughput screening (HTS) is characterized by experimentation on a large or even ultra-large (&gt; 100,000 experiments) scale, which is achieved through a high degree of automation with regards to pipetting and microtiter plate handling as well as through miniaturization of experimental procedures. 
Data generated in HTS are subject to variation caused by different sources and therefore requires appropriate normalization. In general, the following steps are taken:</p>

<ol>
  <li><strong>Input Data:</strong> The columns of the input data need to be assigned correctly, for HiTSeekR to proceed with the analysis.</li>
  <li><strong>Quality Control:</strong> Sources of variation are identified and the quality of replicates and controls are assessed.</li>
  <li><strong>Normalization:</strong> Bias in the data is countered by a suitable normalization method. Control based and plate based normalization methods are differentiated.</li>
  <li><strong>Hit Discovery:</strong> A hit detection method, either based on controls or on plate statistics, is used to identify active samples in the screen.</li>
  <li><strong>Down-stream analysis:</strong> The hit list generated in the step above is used as input for further analyses based on systems biology methods. The goal here is to learn what biological processes are affected rather than relying on individual observations that can also be due to outliers or off-target effects of the samples.</li>
</ol>

<p>HiTSeekR guides the user through each of these steps and enables well informed decisions based on informative plots and direct method comparisons. 
One decisive advantage of the HTS technology is its versatility, i.e. almost any kind of cell based assay can in theory be adapted. 
Since the early days of the technology, where it was mainly used for small molecule screens (drug screens) in pharmaceutical companies, this flexibility has opened up new use cases, such as
RNA interference (gene silencing) or miRNA screens (inhibitors or mimicks). Each of these screen types demands a different strategy for down-stream analysis. 
All three types of screens are supported in HiTSeekR and due to differences in the downstream analysis, the first step in a HiTSeekR analysis is always to choose a screen type as illustrated here:</p>

<p><img src="/HiTSeekR/images/tutorial/select_screen.png" alt="The choice of the screen type is always the first step in HiTSeekR" title="The choice of the screen type is always the first step in HiTSeekR" /></p>

<p>We will elaborat on all three screen types in this tutorial. However, since the first four steps are identical, they will only be shown exemplary for the first type of screen focused on gene silencing.    </p>

<h2 id="rna-interference--crispr-cas9-screens-genes">RNA interference / CRISPR CAS9 screens (Genes)</h2>
<p><a name="siRNA"></a></p>

<h3 id="step-1-input-data">Step 1: Input Data</h3>

<p>After selecting <em>genes</em> on the initial page, the input data tab is shown:</p>

<p><img src="/HiTSeekR/images/tutorial/input_options.png" alt="For the input data you can choose between demo data sets, uploading your own data, or downloading a dataset via its PubChem Assay ID" title="For the input data you can choose between demo data sets, uploading your own data, or downloading a dataset via its PubChem Assay ID" /></p>

<p>Here, the user is offered three choices:</p>

<ol>
  <li>
    <p>Selecting one of the available demo datasets for this screen type</p>
  </li>
  <li>
    <p>Uploading a new dataset in a comma, semicolon or tab separated format</p>
  </li>
  <li>
    <p>Selecting and downloading a dataset from PubChem via its assay id (AID)</p>
  </li>
</ol>

<p>For now, we will select <em>RNAi screen for vorinostat resistance genes</em> as input. This screen was originally obtained from PubChem (https://pubchem.ncbi.nlm.nih.gov/bioassay/743454). This action will change the view:</p>

<p><img src="/HiTSeekR/images/tutorial/input_options2.png" alt="Options after selecting a screen" title="Options after selecting a gene" /></p>

<p>At the bottom of the page, a table is shown that represents a raw representation of the input data. This allows the user to check if the data was 
read in correctly and to identify the content of each column. 
The user now has to decide whether the measured values need to be log2 transformed (stabilizes variance between lower and higher signal values), 
if the B-score should be calculated (This is a computationally expensive score suitable to counter position related effects), and, most importantly,
the user can also assign columns of the input data to properties in HiTSeekR. This step can be omitted for demo data since the column assignment is 
pre-configured. For custom uploaded and PubChem downloaded screens, however, this step is necessary before the analysis can be continued:</p>

<p><img src="/HiTSeekR/images/tutorial/input_options3.png" alt="Assigning columns in the raw data" title="Assigning columns in the raw data" /></p>

<p>Please take
your time to do the column assignment right, because experience shows that most of the problems with using HiTSeekR are caused by incorrect column assignments. A few notes:</p>

<ul>
  <li>For the well position, three different formats are accepted. Row / column indices, alphanumeric well names, e.g. A01, and absoulte well indices, e.g. (1-384)</li>
  <li>For each screen type, we support the most common identifiers, which HiTSeekR will try to map to its default accession type. We advise users to check if this mapping is successful, since outdated identifiers
cannot always be mapped like ambigious miRNA identifiers that do not yet follow the 3p and 5p notation. </li>
  <li>Several measurement, replicate and experiment columns can be selected. The background of this is that in many data sets this information is not fully normalized. This header, for example is not normalized:</li>
</ul>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Sample</th>
      <th style="text-align: center">Experiment1</th>
      <th style="text-align: center">Experiment2</th>
      <th style="text-align: center">Experiment3</th>
      <th style="text-align: center">Accession</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
    </tr>
  </tbody>
</table>

<p>In contrast, this header is normalized:</p>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Sample</th>
      <th style="text-align: center">Experiment</th>
      <th style="text-align: center">Accession</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center">…</td>
      <td style="text-align: center">Experiment1</td>
      <td style="text-align: center">…</td>
    </tr>
    <tr>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
    </tr>
    <tr>
      <td style="text-align: center">…</td>
      <td style="text-align: center">Experiment2</td>
      <td style="text-align: center">…</td>
    </tr>
    <tr>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
      <td style="text-align: center">…</td>
    </tr>
    <tr>
      <td style="text-align: center">…</td>
      <td style="text-align: center">Experiment3</td>
      <td style="text-align: center">…</td>
    </tr>
  </tbody>
</table>

<p>If multiple columns of experiments, readouts, or measurements are selected, HiTSeekR will attempt to construct a normalized table. 
Note that this multiple column feature is currently restricted to one of those three types.
After choosing the appropriate columns the user is encouraged to click on <em>Process raw data</em> to continue. After successful processing, new options will appear in the tab bar on top of the page. 
In addition, continue buttons like the one highlighted here are supposed to guide the user through the process:</p>

<p><img src="/HiTSeekR/images/tutorial/continue_to_qc.png" alt="After the processing of the raw data, new options become available" title="After the processing of the raw data, new options become available" /></p>

<h3 id="step-2-quality-control">Step 2: Quality Control</h3>

<p>In the next step of the analysis, various plots are produced to illuminate various aspects of the screening quality. The following plot, for instance, gives an impression of the signal distribution over
the entire screen. Each box plots here depicts the signal spread of one plate, allowing it to easily spot differences of plate means and signal variance. Please note the checkbox <em>show help text</em>, which, when clicked, will provide 
additional information on each plot to help the user with the interpretation:</p>

<p><img src="/HiTSeekR/images/tutorial/continue_to_qc.png" alt="Quality control" title="Quality control" /></p>

<p>The following plots are available:</p>

<ul>
  <li>Plate Signal Variation: Whole screen box plots of each plate and replicate, divided into different readouts and experiments.</li>
  <li>Row and Column Effect: Bar plot of avarge signal of each row and column over the entire screen. Helps to spot positional bias.</li>
  <li>Control Signal Spread: Depicts in box plots the signal variation of the controls (if included).</li>
  <li>Control Separability: Depicts strictly standardized mean scores (SSMD, Zhang et al.) scores for assessing the separability of positive and negative controls (if given).</li>
  <li>Replicate Correlation: Pairwise comparison of screening replicates including linear regression and R2 value to assess replicate correlation. </li>
</ul>

<p>The main purpose of the quality control step is to learn about bias in the raw screening data. This allows the user to make informed decisions on the choice of normalization method in the next step.</p>

<h3 id="step-3-normalization">Step 3: Normalization</h3>

<p>In this step, various normalization methods can be compared by investigating their effect on the data. First, the user can select one of the following methods:</p>

<ul>
  <li>centered by mean: Divide each plate by its mean signal value </li>
  <li>centered by median: Divide each plate by its median signal value</li>
  <li>z-score: Subtract for each plate the mean value and divide by the standard deviation</li>
  <li>robust z-score: Subtract for each plate the median value and divide by the median absolute deviation</li>
  <li>percentage of negative control: Use the negative controls as reference point (1).</li>
  <li>normalized percentage inhibition: Use positive and negative control as reference points (positive 100, negative 0).</li>
</ul>

<p>The plots will then be updated to reflect the chose normalization. As an example, we here show the plate signal variation plot again, which, with the selected z-score, guarantees that the plate means are now aligned:</p>

<p><img src="/HiTSeekR/images/tutorial/normalization.png" alt="Normalization" title="Normalization" /></p>

<p>In addition, to a downloadable table with all of the normalized values, the following plots are available to study the effect of a normalization method:</p>

<ul>
  <li>Plate Signal Variation: Same as above</li>
  <li>Whole Screen Scatterplot: Similar to the plate signal variation plot, but shows each sample as individual point. For smaller screens an interactive javascript version is rendered.</li>
  <li>Signal Distribution: This plot shows the signal density plot for each replicate. Here, normalization effects can be assessed very effectively by checking how well replicates agree.</li>
  <li>QQ plot: The qq plot shows if the data is normally distributed. In this case most of the data points will form a line from which only active samples will deviate at the beginning and end.</li>
  <li>Plate Viewer: Here, individual plates can be investigated closely by an interactive scatter plot and heatmap. </li>
  <li>Replicate Correlation: Same as above</li>
</ul>

<p>When the user feels confident with the different normalization methods and their effect, we continue with the identification of active samples, also called hits. </p>

<h3 id="step-4-hit-discovery">Step 4: Hit Discovery</h3>

<p>On this page, the user might feel a bit overwhelmed with the different options at first, so let’s go through them one by one, starting with the select boxes:</p>

<ul>
  <li>Select experiment: The user can select one or seveal experiments to check for hits</li>
  <li>Select readout: The user can select one or several readouts to be included</li>
  <li>Normalization: Here, the normalization should be selected that appeared most appropriate after quality control and the normalization step before. </li>
  <li>Hit discovery method: There are several ways to identify hits:
    <ul>
      <li>Fixed cutoffs: This is the most primitive hit detection method in which the user defines an upper and lower threshold to be considered as hard cutoff.</li>
      <li>Standard deviation: Probably the most common hit detection method in which the general variability of the screening data is used as a measure of activity. A common threshold here is a margin of 3 standard deviations.</li>
      <li>Median absolute deviation: Similar to standard deviation method, but more robust due to the use of the median instead of the mean.</li>
      <li>Inter-quartile range: Another variant in which the difference of the 3rd and 1st quartile are used instead to assess variability.</li>
      <li>SSMD: The strictly standardized mean difference proposed by Zhang et al. is a robust alternative to multiple t-tests between the sample and the controls, which is not affected by the number of replciates / samples. A SSMD value of +-3 is considered good, a SSMD value of +-6 is considered excellent. Naturally, this method only works if replicates are available.</li>
      <li>Bayesian statistics: The probably most sophisticated hit detection method takes plate as well as screen-wide statistics into account as priors to assess sample activity as a posterior probability. It is the only method that allows setting a threshold based on a p-value and works best on raw signal values. It depends on the availablity of negative control samples on each plate.
Similar to the normalization step, we encourage users to try out different strategies and thresholds. Comparing the results will reveal what works best for a particular dataset.</li>
    </ul>
  </li>
  <li>Effect: This is interesting if the user is interested in only one effect direction, e.g. only promotors or suppressors.</li>
  <li>Categories: HiTSeekR depicts each hit with one of the following three categories:
    <ul>
      <li>Promotor (blue): The sample shows increased signal, e.g. proliferation effect</li>
      <li>Suppressor (red): The samples shows decreased signal, e.g. killing effect</li>
      <li>Included (yellow): The sample has been manually included in the hit list by the user</li>
    </ul>
  </li>
</ul>

<p>Below the options panel, a tabular hit list is shown. This table, like all tables in HiTSeekR, can be downloaded (button below the table), sorted after arbitrary columns and searched either on the entire table or on a specific column. </p>

<p>The hits plot tab provides a graphical representation of the screening hits in a scatter plot:</p>

<p><img src="/HiTSeekR/images/tutorial/hits_plot.png" alt="Hits plot" title="Hits plot" /></p>

<p>Finally, the Heatmap is a graphical representation of the entire screen. Hits are depicted by small black arrows. 
This plot allows the user to quickly spot if hits accumulate on a specific plate or a general plate location (e.g. mostly in the first row or column etc.), which might indicate problems with the selected normalization or hit identification strategy:</p>

<p><img src="/HiTSeekR/images/tutorial/heatmap.png" alt="Whole screen heatmap with hits indicated by arrows." title="Whole screen heatmap with hits indicated by arrows." /></p>

<p>When the user has successfully selected hits we are ready for downstream analysis. Therefore, additional tabs are unlocked as soon as the hit discovery tab is selected.</p>

<h3 id="step-5-downstream-analysis">Step 5: Downstream Analysis</h3>

<h4 id="gene-sets">Gene sets</h4>

<h4 id="de-novo-network-enrichment">De novo network enrichment</h4>

<h2 id="microrna-inhibitor--mimicks-screens">microRNA inhibitor / mimicks screens</h2>
<p><a name="miRNA"></a></p>

<h3 id="steps-1-4">Steps 1-4</h3>

<p>See RNA interference.</p>

<h3 id="step-5-downstream-analysis-1">Step 5: Downstream Analysis</h3>

<h2 id="small-compound-drug-screen">Small compound (drug) screen</h2>
<p><a name="compound"></a></p>

<h3 id="steps-1-4-1">Steps 1-4</h3>

<p>See RNA interference</p>

<h3 id="step-5-downstream-analysis-2">Step 5: Downstream Analysis</h3>

          </div>
          <div class="col-sm-4">
            <!-- Feel free to remove this div. Uses purplecoat: http://ellekasai.github.io/purplecoat.js -->
  <h3 class="section-header">
    Get started:
  </h3>
  <p><a href="http://www.hitseekr.combio.sdu.dk" target="_blank" class="btn btn-primary">Demo Application</a></p>

<h3>
  Tutorial:
</h3>
<ul>
	<li>
		<a href="/HiTSeekR/tutorial/index.html#Introduction" >Introduction</a>
	</li>
	<li>
		<a href="/HiTSeekR/tutorial/index.html#siRNA" >Example 1: siRNA screen</a>
	</li>
	<li>
		<a href="/HiTSeekR/tutorial/index.html#miRNA" >Example 2: microRNA screen</a>
	</li>
	<li>
		<a href="/HiTSeekR/tutorial/index.html#compound" >Example 3: drug screen</a>
	</li>
</ul>

<h3>
  Setup your own installation:
</h3>
	<p><a href="https://hub.docker.com/r/nanocan/hitseekr/" target="_blank" class="btn btn-primary">Docker Hub</a></p>

<h3>
  <h3 class="section-header">
    Get involved:
  </h3>

  <p><a href="https://github.com/nanocan/HitSeekR" target="_blank" class="btn btn-primary"><i class="fa fa-github fa-lg"></i> View on GitHub &rarr;</a></p>


  <h3>
    Design & Development: 
  </h3>
  <h4>Markus List</h4>

  <p><a href="https://twitter.com/itisalist" class="twitter-follow-button" data-show-screen-name="false" data-show-count="true" data-dnt="true" data-size="large">Follow @itisalist</a></p>

<h3>
	Supported by:
</h3>          
	<div><a href='http://nanocan.org'><img width=150 src="/HiTSeekR/images/NanoCAN.png"/></a></div>
   <div><a href='http://baumbachlab.net'><img width=200 src="/HiTSeekR/images/baumbachlab.png"/></a></div>
 


          </div>
        
      </div>
      <div class="row footer">
        <div class="col-sm-12 text-center">
          <!-- Feel free to remove this div. Uses purplecoat: http://ellekasai.github.io/purplecoat.js -->
<div data-purplecoat="shiori" data-purplecoat-label="footer.html">
&copy;2015.
Built with <a href="http://jekyllrb.com/">Jekyll</a> and
<a href="https://github.com/ellekasai/shiori">Shiori Theme</a>.
</div>
        </div>
      </div>
    </div>
    <script src="/HiTSeekR/javascripts/jquery.min.js"></script>
    <script src="/HiTSeekR/javascripts/bootstrap.min.js"></script>
    <script src="/HiTSeekR/javascripts/purplecoat-min.js"></script>
    <!-- Place your <script> tags here. -->

<!-- Example: Twitter -->
<script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+'://platform.twitter.com/widgets.js';fjs.parentNode.insertBefore(js,fjs);}}(document, 'script', 'twitter-wjs');</script>

  </body>
</html>
